pkg/asset/machines/gcp/machines.go

```diff
@@ -79,7 +79,7 @@ func provider(clusterID string, platform *gcp.Platform, mpool *gcp.MachinePool,
 			Boot:   true,
 			SizeGb: 128,
 			Type:   "pd-ssd",
-			Image:  osImage,
+			Image:  fmt.Sprintf("%s-rhcos-image", clusterID),
```

Comment:
Shouldn't we include e.g. the OpenShift major version here, like `4.1`, `4.2` etc?

Reply:
As it stands today, we only have a single bootimage for all versions.. so this is appropriate.

Reply:
Namespacing with the cluster ID I think in general is OK.  But the reason I'm advocating including the version number is around issues stemming from https://github.com/openshift/os/issues/381

My concern is:

 - User sets up 4.2 cluster in GCP, cluster stays running
 - 4.3 is released
 - User later runs 4.3 installer which will have a new bootimage - if we end up overwriting that boot image, would e.g. newly scaled workers in the 4.2 cluster start using it? 

Basically I'm not sure we really want to think about supporting/dealing with 4.X clusters booting from 4.(X-1) etc.

Again I think the cluster ID here is OK, but think about e.g. bare metal PXE setups etc. where we don't programatially control the bootimages and may have multiple clusters at the same time; we want to encourage people in general to "version namespace" the bootimages.

Reply:
> Namespacing with the cluster ID I think in general is OK. But the reason I'm advocating including the version number is around issues stemming from [openshift/os#381](https://github.com/openshift/os/issues/381)
> 
> My concern is:
> 
> * User sets up 4.2 cluster in GCP, cluster stays running
> * 4.3 is released
> * User later runs 4.3 installer which will have a new bootimage - if we end up overwriting that boot image, would e.g. newly scaled workers in the 4.2 cluster start using it?

The infra is is randomized for that purpose. So 2 clusters will have their own image.

> Basically I'm not sure we really want to think about supporting/dealing with 4.X clusters booting from 4.(X-1) etc.
> 
> Again I think the cluster ID here is OK, but think about e.g. bare metal PXE setups etc. where we don't programatially control the bootimages and may have multiple clusters at the same time; we want to encourage people in general to "version namespace" the bootimages.

I would suggest even those uses to create bootimage locations to be specific to the cluster.. they have access to infra-id

And as for versions in names, I don't think that are necessary here.